libs/mock-block-dock/src/datastore/traverse.ts

```diff
@@ -1,284 +1,543 @@
 
-              addKnowledgeGraphEdgeToSubgraphByMutation(
-                traversalSubgraph,
-                entityId,
-                edgeTimestamp,
-                outgoingLinkEdge,
-              );
-
-              /** @todo - This is temporary, and wrong */
-              toResolve.insert(
-                {
-                  baseId: outgoingLinkEntity.metadata.recordId.entityId,
-                  revisionId: outgoingLinkEntity.metadata.recordId.editionId,
-                },
-                {
-                  [edgeKind]: { incoming: depths.incoming - 1 },
-                },
-              );
-            }
-          }
-          if (depths.outgoing) {
-            if (
-              "linkData" in element.inner &&
-              element.inner.linkData?.leftEntityId !== undefined &&
-              element.inner.linkData?.rightEntityId !== undefined
-            ) {
-              // get left entity for link entity and insert edges
-              const leftEntity = getLeftEntityForLinkEntity(
-                datastore,
-                entityId,
-              );
-              const {
-                metadata: {
-                  recordId: {
-                    entityId: leftEntityId,
-                    editionId: edgeTimestamp,
-                  },
-                },
-              } = leftEntity;
+        if (isEntityVertex(neighborVertex)) {
+          // get from temporal data of the neighbor vertex
+          const entityValidInterval =
+            neighborVertex.inner.metadata.temporalVersioning[
+              traversalSubgraph.temporalAxes.resolved.pinned.axis
+            ];
+          newIntersection = intervalIntersectionWithInterval(
+            interval,
+            entityValidInterval,
+          );
+          neighborVertexId = {
+            baseId: neighborVertex.inner.metadata.recordId.entityId,
+            revisionId: entityValidInterval.start.limit,
+          };
+        } else {
+          newIntersection = interval;
+          neighborVertexId = {
+            baseId: neighborVertex.inner.metadata.recordId.baseUri,
+            revisionId: neighborVertex.inner.metadata.recordId.version,
+          };
+        }
 
-              const hasLeftEntityEdge: HasLeftEntityEdge = {
-                kind: "HAS_LEFT_ENTITY",
-                reversed: false,
-                rightEndpoint: {
-                  baseId: leftEntityId,
-                  timestamp: edgeTimestamp,
-                },
-              };
+        if (newIntersection) {
+          traverseElementTemporal({
+            traversalSubgraph,
+            datastore,
+            element: neighborVertex,
+            elementIdentifier: neighborVertexId,
+            interval: newIntersection,
+            currentTraversalDepths: {
+              ...currentTraversalDepths,
+              [edgeKind]: {
+                ...currentTraversalDepths[edgeKind],
+                [direction]: depth - 1,
+              },
+            },
```

Comment:
nit: I don't know what's common in TS, but I prefer to first write the changed values, then the fallback, so something like
```suggestion
            currentTraversalDepths: {
              [edgeKind]: {
                [direction]: depth - 1,
                ...currentTraversalDepths[edgeKind],
              },
              ...currentTraversalDepths,
            },
```

Reply:
That would result in the fields being overridden https://stackoverflow.com/a/63238111/14687716

Reply:
Well, this would be suboptimal 🙃